src/core/trtserver.h

```diff
@@ -125,22 +125,40 @@ TRTSERVER_EXPORT const char* TRTSERVER_ErrorMessage(TRTSERVER_Error* error);
 /// maintains a reference into the block.
 ///
 
-/// Create a new shared memory block object referencing a shared
+/// Create a new shared memory block object referencing a system shared
 /// memory block residing in TRTSERVER_MEMORY_CPU type memory.
 /// \param shared_memory_block Returns the new shared memory block object.
 /// \param name A unique name for the shared memory block. This name
 /// is used in inference requests to refer to this shared memory
 /// block.
-/// \param shm_key The name of the posix shared memory object
+/// \param shm_key The name of the posix system shared memory object
 /// containing the block of memory.
-/// \param offset The offset within the shared memory object to the
+/// \param offset The offset within the system shared memory object to the
 /// start of the block.
 /// \param byte_size The size, in bytes of the block.
 /// \return a TRTSERVER_Error indicating success or failure.
 TRTSERVER_EXPORT TRTSERVER_Error* TRTSERVER_SharedMemoryBlockCpuNew(
     TRTSERVER_SharedMemoryBlock** shared_memory_block, const char* name,
     const char* shm_key, const size_t offset, const size_t byte_size);
 
+/// Create a new shared memory block object referencing a CUDA shared
+/// memory block residing in TRTSERVER_MEMORY_GPU type memory.
+/// \param shared_memory_block Returns the new shared memory block object.
+/// \param name A unique name for the shared memory block. This name
+/// is used in inference requests to refer to this shared memory
+/// block.
+/// \param shm_name The name of the system shared memory block containing
+/// the CUDA IPC handle.
+/// \param offset The offset within the CUDA shared memory object to the
```

Comment:
These comments are wrong or else something is wrong elsewhere. We should discuss.

Reply:
Fixed. It is the offset to the start system shared memory block to the start of the CUDA shared memory object.

Reply:
Edit: Split into TRTSERVER_SharedMemoryBlockGpuNew and TRTSERVER_SharedMemoryBlockCpuNew to improve understandably